test/constructorio-synonyms.js

```diff
@@ -0,0 +1,648 @@
+/* eslint-disable prefer-destructuring, no-unused-expressions, no-console, max-nested-callbacks */
+
+const expect = require('chai').expect;
+const Constructorio = require('../lib/constructorio');
+
+const testConfig = {
+  apiToken: 'YSOxV00F0Kk2R0KnPQN8',
+  apiKey: 'ZqXaOfXuBWD4s3XzCI1q',
+};
+
+// Helper method - create synonym group with random synonyms
+function addTestSynonymGroup() {
+  const constructorio = new Constructorio(testConfig);
+
+  return new Promise((resolve, reject) => {
+    constructorio.addSynonymGroup({
+      synonyms: [
+        `${Math.random().toString(36).substring(2, 15)}`,
+        `${Math.random().toString(36).substring(2, 15)}`,
+      ],
+    }, (err, response) => {
+      if (err) {
+        return reject(err);
+      }
+
+      return resolve(response);
+    });
+  });
+}
+
+// Helper method - remove synonym group for specified id
+function removeTestSynonymGroup(id) {
+  const constructorio = new Constructorio(testConfig);
+
+  return new Promise((resolve, reject) => {
+    constructorio.removeSynonymGroup({
+      group_id: id,
+    }, (err, response) => {
+      if (err) {
+        return reject(err);
+      }
+
+      return resolve(response);
+    });
+  });
+}
+
+describe('ConstructorIO - Synonym Groups', () => {
+  describe('addSynonymGroup', () => {
+    let addedSynonymGroupId = null;
+
+    after((done) => {
+      // Clean up - remove synonym group created for tests
+      removeTestSynonymGroup(addedSynonymGroupId).then(done).catch((err) => {
```

Comment:
Should we extract the `add/removeTestSynonymGroup` call/warning code out into a function since similar code is ran in the `before` and `after` in other describe blocks?